[docs] [explore] improve the introduction to follower reads page. (#11709)
yugabyte/yugabyte-db: docs/content/latest/explore/ysql-language-features/going-beyond-sql/follower-reads-ysql.md

  </li>
</ul>

## Leader lease mechanism

> In a distributed environment, when one node in a cluster is elected as the leader holding the latest data, it is possible that another node may assume that it is the leader, and that it holds the latest data. This could result in serving stale reads to a client. To avoid this confusion, YugabyteDB provides a **leader lease** mechanism  where an elected node member is guaranteed to be the leader until it's lease expires.

Reviewer: ```suggestion
In a distributed environment, when one node in a cluster is elected as the leader holding the latest data, it is possible that another node may assume that it is the leader, and that it holds the latest data. This could result in serving stale reads to a client. To avoid this confusion, YugabyteDB provides a _leader lease_ mechanism  where an elected node member is guaranteed to be the leader until its lease expires.
```

The leader lease mechanism guarantees to serve strongly consistent reads where a client can fetch reads directly from the leader, because the leader under lease will have the latest data.
The interactive animations in this [blog post](https://blog.yugabyte.com/low-latency-reads-in-geo-distributed-sql-with-raft-leader-leases/) explains the performance improvements using leader leases.

## Introduction to Follower reads

YugabyteDB requires reading from the **leader** to read the latest data. However, for applications that don't require the latest data, or are working with unchanging data, the cost of contacting a potentially remote leader to fetch the data may be wasteful. Your application may benefit from better latency by reading from a **replica** that is closer to the client.

> ### Scenario where stale reads are not acceptable
> 
> Let's say a user starts a donation page to raise money for a personal cause and the target amount must be met by the end of the day. In such a scenario which is time critical, the user will benefit from accessing the most recent donation made on the page and can keep track of the progress. Such real-time applications would require to get the most latest information as soon as it is available. But if there are applications where fetching slightly stale data is okay, then the follower reads serve that purpose.

Reviewer: ```suggestion
### Time-critical use cases

Let's say a user starts a donation page to raise money for a personal cause and the target amount must be met by the end of the day. In a time-critical scenario, the user benefits from accessing the most recent donation made on the page and can keep track of the progress. Such real-time applications require the latest information as soon as it's available. But in the case of an application where fetching slightly stale data is okay, follower reads can be helpful.
```

> ### Scenario where an application can handle stale reads

Reviewer: ```suggestion
### Latency-tolerant use cases
```

> Let's say a social media post gets about a million likes and more continuously. For a post with massive likes as this one, slightly stale reads are acceptable and it does not require immediate updates because the number may not really matter to the user reading the post. Such applications don't require to always process requests from the leader. Instead, a slightly older value from the closest replica can achieve improved performance with low latency.

Reviewer: ```suggestion
Let's say a social media post gets about a million likes and more continuously. For a post with massive likes such as this one, slightly stale reads are acceptable, and immediate updates aren't necessary because the absolute number may not really matter to the user reading the post. Such applications don't need to always make requests directly to the leader. Instead, a slightly older value from the closest replica can achieve improved performance with lower latency.
```

> Follower reads are applicable for applications that can tolerate staleness.
> Replicas may not be completely up to date with all updates, so this design may respond with stale data. You can specify how much staleness the application can tolerate. When enabled, read-only operations may be handled by the closest replica, instead of having to go to the leader. The GUC session variables that PostgreSQL supports can be used to enable follower reads.

Reviewer: Do we need (or want) to link to a reference for GUC variables, or should we just leave it as-is?

```suggestion
Follower reads are applicable for applications that can tolerate staleness. Replicas may not be completely up to date with all updates, so this design may respond with stale data. You can specify how much staleness the application can tolerate. When enabled, read-only operations may be handled by the closest replica, instead of having to go to the leader. The GUC session variables that PostgreSQL supports can be used to enable follower reads.
```
